ember-server/src/test/scala/org/http4s/ember/server/EmberServerSuite.scala

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2019 http4s.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.http4s.ember.server
+
+import cats.syntax.all._
+import cats.effect._
+import org.http4s._
+import org.http4s.client.Client
+import org.http4s.server.Server
+import org.http4s.implicits._
+import org.http4s.dsl.Http4sDsl
+import org.http4s.ember.client.EmberClientBuilder
+
+import scala.concurrent.duration._
+
+class EmberServerSuite extends Http4sSuite {
```

Comment:
You'll notice that I didn't actually include any tests for the graceful shutdown mechanism... Unsurprisingly that behavior is rather hard to exercise since there's a lot of nondeterministic elements to it. In particular, we can't deterministically start a request _after_ the server socket channel has been closed.

Reply:
I think we could batter it with requests and assert that every request either completed successfully _or_ was refused a connection.  But our CI is suffering enough as it is.